torch/_inductor/kernel/flex_decoding.py

```diff
@@ -0,0 +1,363 @@
+
+    # Store output, logsumexp and rowmax for cross CTA reduction. (all in float32, even when input data are in fp16)
+    if SAFE_M_BOUNDARY:
+        tl.store(M_block_ptr, m_i[None, :])
+        tl.store(L_block_ptr, l_i[None, :])
+    else:
+        tl.store(M_block_ptr, m_i[None, :], boundary_check=(1,))
+        tl.store(L_block_ptr, l_i[None, :], boundary_check=(1,))
+
+    # -- store output
+    idx_z = off_z
+    idx_h = off_h
+    idx_t = off_t
+    idx_m = offs_m[:, None]
+    idx_d = offs_d[None, :]
+    # TODO generalize and add proper mask support
+    mask = (idx_m < Q_CTX) & (idx_d != -1)
+    {{store_output(("idx_z", "idx_h", "idx_t", "idx_m", "idx_d"), "acc", "mask")}}
+ """,
+)
+
+
+MAX_SPLIT_KV = 64
+
+
+def get_split_k(B: int, H: int, Mk: int, SM: int = 128) -> int:
+    """Heuristic for the number of splits from xformer"""
+    bh = max(B * H, 1)  # NOTE: Handle B*h=0 case
+    split_k = SM // bh  # Each SM should at least get one block.
+    split_k = max(split_k, 1)
+
+    return split_k
+
+
+def _get_decoding_default_config(key) -> Tuple[int, int, int]:
+    B = key.get_size()[0]
+    H = key.get_size()[1]
+    Mk = key.get_size()[-2]
+    head_dim = key.get_size()[-1]
+    dtype = key.get_dtype()
+    default_config = None
+
+    default_config = (64, 2, 3)
+
+    return default_config
+
+
+def create_flex_decoding_kernel(*args, **kwargs):
+    (subgraph_buffer, query, key, value, subgraph, *other_buffers) = args
+    choices: List[Any] = []
+    configs: List[Tuple[int, int, int]] = []
+    configs.append(_get_decoding_default_config(key))
+    # Note: max_autotune is not supported yet. Causes error in lowering the dynamic shape in reduction ops.
+    # if config.max_autotune:
+    #     configs += [
+    #         (64, 2, 2),
+    #         (32, 2, 3),
+    #     ]
+
+    SPLIT_KV = get_split_k(key.get_size()[0], key.get_size()[1], key.get_size()[2])
+    MAX_SPLIT_KV = SPLIT_KV
+    assert SPLIT_KV <= MAX_SPLIT_KV
+
+    # create config dependent intermediate buffers
+    buf_ML_shape = query.get_size()[:-2] + [
+        MAX_SPLIT_KV,
+        query.get_size()[-2],
+    ]  # [B, H, SPLIT_KV, M]
+    buf_M = empty_strided(
+        buf_ML_shape,
+        None,
+        dtype=torch.float32,  # The rowmax is always stored in fp32 regardless of the input dtype
+        device=query.get_device(),
+    )
+    buf_L = empty_strided(
+        buf_ML_shape,
+        None,
+        dtype=torch.float32,  # The intermediate sumexp is always stored in fp32 regardless of the input dtype
+        device=query.get_device(),
+    )
+
+    buf_ACC_shape = (
+        query.get_size()[:-2] + [MAX_SPLIT_KV] + query.get_size()[-2:]
+    )  # [B, H, SPLIT_KV, M, D]
+
+    layout_acc = FixedLayout(
+        query.get_device(),
+        torch.float32,
+        buf_ACC_shape,
+        FlexibleLayout.contiguous_strides(buf_ACC_shape),
+    )
+
+    BLOCK_M = (
+        query.get_size()[-2]
+        if query.get_size()[-2] <= 0
```

Comment:
add a note that this is intentional.

Reply:
This is a works around before Triton fix https://github.com/triton-lang/triton/pull/4061 is in pin.